src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RuntimeCSharpRenderer.cs

```diff
@@ -210,6 +210,7 @@ public override void VisitCSharpAttributeValue(CSharpAttributeValueIRNode node)
 
         public override void VisitCSharpStatement(CSharpStatementIRNode node)
         {
+            // We can't remove this yet, because it's still used recursively in a few places.
```

Comment:
Did you validate this? Did you try it? What kinds of things had statements nested inside of them?